sample/src/main/java/com/github/barteksc/sample/PDFViewActivity.java

```diff
@@ -223,4 +252,173 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String permissi
     public void onPageError(int page, Throwable t) {
         Log.e(TAG, "Cannot load page " + page);
     }
+
+    @Override
+    public void onLayerDrawn(Canvas canvas, float pageWidth, float pageHeight, int displayedPage) {
+        // Draw in progress annotation (currently drawn by user)
+        if (mAnnotationDrawingPath != null) {
+            Paint paint1 = new Paint();
+            paint1.setStyle(Paint.Style.STROKE);
+            paint1.setStrokeWidth(10);
+            paint1.setColor(Color.BLACK);
+            canvas.drawPath(mAnnotationDrawingPath, paint1);
+        }
+
+        // Draw cached annotations
+        Paint paint2 = new Paint();
+        paint2.setStyle(Paint.Style.STROKE);
+        paint2.setStrokeWidth(10);
+        paint2.setColor(Color.RED);
+
+        Log.d("OnDrawLayer", "############ ANNOTATIONS #################");
```

Comment:
that's not spamming the device log on every draw is it? I guess .d won't do this on the actual device?

Reply:
This is just sample app to test if all changes are working as expected